src/github.com/getlantern/flashlight/main/pac.go

```diff
@@ -20,22 +21,32 @@ import (
 )
 
 var (
-	isPacOn        = int32(0)
-	pacURL         string
-	muPACFile      sync.RWMutex
-	pacFile        []byte
-	directHosts    = make(map[string]bool)
-	shouldProxyAll = int32(0)
-	onepac         = &sync.Once{}
+	isPacOn     = int32(0)
+	pacURL      string
+	directHosts = make(map[string]bool)
+	cfgMutex    sync.RWMutex
 )
 
-func ServeProxyAllPacFile(b bool) {
-	if b {
-		atomic.StoreInt32(&shouldProxyAll, 1)
-	} else {
-		atomic.StoreInt32(&shouldProxyAll, 0)
+func ServePACFile() {
+	cfgMutex.Lock()
+	defer cfgMutex.Unlock()
+	servePACFileIfNecessary()
+}
+
+func servePACFileIfNecessary() {
+	if pacURL == "" {
+		handler := func(resp http.ResponseWriter, req *http.Request) {
+			log.Trace("Serving PAC file")
+			resp.Header().Set("Content-Type", "application/x-ns-proxy-autoconfig")
+			resp.WriteHeader(http.StatusOK)
+			cfgMutex.RLock()
+			defer cfgMutex.RUnlock()
+			if _, err := genPACFile(resp); err != nil {
```

Comment:
Yes, this means we're generating the PAC file dynamically every time.  No it's not a performance problem, because the browser infrequently requests the PAC file.  Also, a boom test reveals that we can serve the dynamic file at about 20,000 per second.